### PR TITLE
Bugfix/144360 possible to open documents by url when user not access

### PR DESCRIPTION
**Pull request by pablocodeglia**
- Created new functional guards to check for Innovation Status and User Role.
- Implemented checkStatusGuard to 'documents' and 'support-summary' on Innovator's routing module.

Closes [AB#144360](https://dev.azure.com/NHSiDev/InnovatorService/_workitems/edit/144360).

### CODE REVIEW

**Review by eduarda-meirinhos-bjss — APPROVED** [review]

File: src/modules/feature-modules/innovator/guards/check-status.guard.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { of } from 'rxjs';
+
+export function checkStatusGuard(statusList: InnovationStatusEnum[], blockList: boolean = false): CanActivateFn {
+  return (route: ActivatedRouteSnapshot) => {
+    const router: Router = inject(Router);
+    const contextService: ContextService = inject(ContextService);
+    const authenticationStore: AuthenticationStore = inject(AuthenticationStore);
+
+    return contextService
+      .getInnovationContextInfo(route.params.innovationId, authenticationStore.getUserContextInfo())
```
Comment (diogo-costa-bjss) [inline src/modules/feature-modules/innovator/guards/check-status.guard.ts:15]: I think you should be using the `getOrLoadInnovation` from context store here

File: src/modules/stores/context/context.store.ts [hunk 2]
```diff
@@ -188,6 +189,22 @@ export class ContextStore extends Store<ContextModel> {
...
+    context: AuthenticationModel['userContext']
+  ): Observable<ContextInnovationType> {
+    if (this.state.innovation) {
+      return of(this.state.innovation);
+    }
+
+    return this.contextService.getInnovationContextInfo(innovationId, context).pipe(
+      tap(innovation => {
+        this.setInnovation(innovation);
+        return innovation;
```
Comment (diogo-costa-bjss) [inline src/modules/stores/context/context.store.ts:203]: I am unsure, but I don't think `tap` needs to return anything.

**Review by diogo-costa-bjss — CHANGES_REQUESTED** [review]

File: src/modules/stores/context/context.store.ts (hunk 2 above)
Comment (pablocodeglia) [reply src/modules/stores/context/context.store.ts]: Tested it and yeah, no return needed indeed, nice one

File: src/modules/feature-modules/innovator/guards/check-status.guard.ts (hunk 1 above)
Comment (pablocodeglia) [reply src/modules/feature-modules/innovator/guards/check-status.guard.ts]: Right, correct. Changed it as suggested

**Review by diogo-costa-bjss — APPROVED** [review]

**Pull request merged**